text/0028-ed25519-validation/0028-ed25519-validation.md

```diff
@@ -38,19 +38,23 @@ The Curve25519 is defined over the finite field of order p=2<sup>255</sup>−19.
+In contrast to RFC 8032, it is _not_ required that the encodings of A and R are canonical. As long as the corresponding (x,y) is a valid curve point, any of such edge cases will be accepted.
 
 ## Validation 
 
 The RFC 8032 mentions two alternative verification equations:
 1. [8][S]B = [8]R + [8][k]A'
 2. [S]B = R + [k]A'
 
-Each honestly generated signature following RFC 8032 satisfies the second, cofactorless equation and thus, also the first equation. However, the opposite is not true: There are solutions only satisfying the first but not the latter.<br> This ambiguity in RFC 8032 has led to the current situation in which different implementations rely on different verification equations. 
+Each honestly generated signature following RFC 8032 satisfies the second, cofactorless equation and thus, also the first equation. However, the opposite is not true: There are solutions only satisfying the first but not the latter. This ambiguity in RFC 8032 has led to the current situation in which different implementations rely on different verification equations. 
 
-Ed25519 also supports batch signature verification, which allows verifying several signatures in a single step, much faster than verifying signatures one-by-one. Correspondingly, there are also two alternative verification equations for the batch verification: cofactored and cofactorless. However, only cofactored verifications, single and batch, are compatible with each other. All other combinations are inconsistent and can lead to false positives or false negatives (see [Chalkias et al. 2020](https://eprint.iacr.org/2020/1244), Section 3.2).
 
-Thus, in order to allow batch signature verification and its faster performance in IOTA nodes, the cofactored version _must_ be used for validation, i.e. the group equation [8][S]B = [8]R + [8][k]A' for the single verification.
+Ed25519 also supports batch signature verification, which allows verifying several signatures in a single step, much faster than verifying signatures one-by-one. Without going into detail, there are also two alternative verification equations for the batch verification:
+[8][∑z<sub>i</sub>s<sub>i</sub>] B = [8]∑[z<sub>i</sub>]R<sub>i</sub> + [8]∑[z<sub>i</sub>h<sub>i</sub>]A<sub>i</sub> and its corresponding cofactorless version. However, only cofactored verifications, single and batch, are compatible with each other. All other combinations are inconsistent and can lead to false positives or false negatives (see [Chalkias et al. 2020](https://eprint.iacr.org/2020/1244), Section 3.2).<br> Thus, in order to allow batch signature verification and its faster performance in IOTA nodes, the cofactored version _must_ be used for validation, i.e. the group equation [8][S]B = [8]R + [8][k]A' for the single verification.
```

Comment:
>its corresponding cofactorless version
Cofactorless version doesn't work (works incorrectly with high probability) with mixed-order points which we have to support for compatibility and etc. So there's only one suitable version of batch equation.
Cofactorless single and batch equations are consistent only in case where points are of prime order which we can't have.

Reply:
I honestly like the current version better. We don't have mentioned "mixed-order" and all this stuff anywhere yet. And I guess I'd rather leave it that way to keep the RFC concise. How about the following compromise:

However, only cofactored verifications, single and batch, are compatible with each other. All other combinations are inconsistent and can lead to false positives or false negatives (see Chalkias et al. 2020, Section 3.2) **for certain edge-case that can be introduced by an attacker.**